### PR TITLE
feat(STONEINTG-674): update kubebuilder to v4

### PR DESCRIPTION
**Pull request by hongweiliu17**
update kubebuilder to v4 according to https://book.kubebuilder.io/migration/manually_migration_guide_gov3_to_gov4
* move main.go to cmd folder
* rename contollers folder to controller under  new internal folder
* update Makefile
* update Dockerfile
* update admissionv1beta1 to admissionv1
* update PROJEDT
## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: api/v1beta2/integrationtestscenario_suite_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
-	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	admissionv1beta1 "k8s.io/api/admission/v1"
```
Comment (MartinBasti) [inline api/v1beta2/integrationtestscenario_suite_test.go:34]: this is beta version, it's just v1, do we need this change?

**Review by MartinBasti — APPROVED** [review]

**Review by dirgim — APPROVED** [review]
LGTM, just needs a rebase

**Review by Josh-Everett — APPROVED** [review]

**Comment by dheerajodha** [comment]
@hongweiliu17 why are we moving all of the controllers within a new "internal" folder? commit message is missing the description

**Comment by hongweiliu17** [comment]
> @hongweiliu17 why are we moving all of the controllers within a new "internal" folder? maybe add explanation to commit message

Added. Thanks.

**Review by dheerajodha — APPROVED** [review]

**Pull request merged**